src/tallies/MeshTally.C

```diff
@@ -144,34 +138,41 @@ MeshTally::spatialFilter()
   // Create the OpenMC mesh which will be tallied on.
   if (!_mesh_template_filename)
   {
+    auto msh = dynamic_cast<const libMesh::ReplicatedMesh *>(_mesh.getMeshPtr());
+    if (!msh)
+      mooseError("Internal error: The mesh is not a replicated mesh.");
```

Comment:
Would this ever happen? I thought we had checks elsewhere that make sure the mesh is not distributed. Just wondering if this could catch some other error or if it is redundant.

Reply:
This is a redundant check. It shouldn't happen, I just have it here in case something changes in libMesh (they add another derived ReplicatedMesh equivalent or something along those lines) to make debugging less painful.